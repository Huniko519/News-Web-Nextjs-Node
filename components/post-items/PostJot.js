import React from 'react';
import PropTypes from 'prop-types';
import PostItemBadge from './PostItemBadge';
import PostItemTitle from './PostItemTitle';
import PostItemLabel from './PostItemLabel';
import PostItemPubDate from './PostItemPubDate';
import Qoute from '../../utils/Qoute';

/**
 * Post jots show the child category and post headline text. No images
 * Will optionally show the badge prepeneded to the headline
 *
 * ╔════════════╗
 * ║ Category   ║
 * ║            ║
 * ║ Headline   ║
 * ╚════════════╝
 *
 * @param {post: {...postData}} props
 */
const PostJot = (props) => {
  const {
    post, showAuthor, showPubDate, showCategory,
  } = props;

  const isShowAuthor = (showCategory === true) ? false : !!((showAuthor || post.template === 'opinion'));

  return (
    <div
      className={`inews__post inews__post-jot ${post.template ? post.template : ''}`}
      key={post.id}
      data-post-id={post.id}
    >

      <div className="inews__post-jot__content">
        <PostItemBadge post={post} showAuthor={isShowAuthor} />
        <div className="inews__post-jot__content-headline">
          {post.template === 'opinion' && <Qoute size="small" />}
          <PostItemLabel post={post} />
          <PostItemTitle post={post} />
        </div>
        <PostItemPubDate post={post} showPubDate={showPubDate} />
      </div>
    </div>
  );
};

/* istanbul ignore next */
PostJot.propTypes = {
  post: PropTypes.objectOf(PropTypes.any).isRequired,
  showAuthor: PropTypes.bool,
  showPubDate: PropTypes.bool,
  showCategory: PropTypes.bool,
};

/* istanbul ignore next */
PostJot.defaultProps = {
  showAuthor: false,
  showPubDate: false,
  showCategory: false,
};

export default PostJot;
