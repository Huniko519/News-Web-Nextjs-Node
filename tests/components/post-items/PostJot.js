import React from 'react';
import PostJot from '../../../components/post-items/PostJot';

beforeEach(() => {
  jest.clearAllMocks();
});

test('renders the snapshot', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postJot = shallow(<PostJot post={post} />);
  expect(postJot).toMatchSnapshot();
});

it('should output with the correct class on the root element', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postJot = shallow(<PostJot post={post} />);
  expect(postJot.hasClass('inews__post')).toBe(true);
  expect(postJot.hasClass('inews__post-jot')).toBe(true);
});

it('should call the <PostItemTitle /> component with the correct props', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };

  const postJot = shallow(<PostJot post={post} />);
  const postItemMedia = postJot.find('PostItemTitle');
  expect(postItemMedia.length).toBe(1);
  expect(postItemMedia.prop('post')).toMatchObject(post);
});

it('should output a <PostItemBadge /> component', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };
  const postJot = shallow(<PostJot post={post} />);
  expect(postJot.find('PostItemBadge').length).toBe(1);
  expect(postJot.find('PostItemBadge').prop('showAuthor')).toBe(false)
  expect(postJot.find('PostItemBadge').prop('post')).toMatchObject(post);
});

it('should not output the <PostItemMedia />', () => {
  const post = {
    id: 1,
    title: {
      rendered: 'title',
    },
    breadcrumbs: [
      { name: 'News' },
    ],
    link: 'https://inews-article.com/test-article',
  };

  const postJot = shallow(<PostJot post={post} />);
  const postItemMedia = postJot.find('PostItemMedia');
  expect(postItemMedia.length).toBe(0);
});
