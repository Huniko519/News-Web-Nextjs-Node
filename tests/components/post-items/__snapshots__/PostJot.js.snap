// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the snapshot 1`] = `
<div
  className="inews__post inews__post-jot "
  data-post-id={1}
  key="1"
>
  <div
    className="inews__post-jot__content"
  >
    <PostItemBadge
      post={
        Object {
          "breadcrumbs": Array [
            Object {
              "name": "News",
            },
          ],
          "id": 1,
          "link": "https://inews-article.com/test-article",
          "title": Object {
            "rendered": "title",
          },
        }
      }
      showAuthor={false}
    />
    <div
      className="inews__post-jot__content-headline"
    >
      <PostItemLabel
        post={
          Object {
            "breadcrumbs": Array [
              Object {
                "name": "News",
              },
            ],
            "id": 1,
            "link": "https://inews-article.com/test-article",
            "title": Object {
              "rendered": "title",
            },
          }
        }
      />
      <PostItemTitle
        post={
          Object {
            "breadcrumbs": Array [
              Object {
                "name": "News",
              },
            ],
            "id": 1,
            "link": "https://inews-article.com/test-article",
            "title": Object {
              "rendered": "title",
            },
          }
        }
      />
    </div>
    <PostItemPubDate
      post={
        Object {
          "breadcrumbs": Array [
            Object {
              "name": "News",
            },
          ],
          "id": 1,
          "link": "https://inews-article.com/test-article",
          "title": Object {
            "rendered": "title",
          },
        }
      }
      showPubDate={false}
    />
  </div>
</div>
`;
